### ISSUE TITLE
[hub] - hf hub internal api no longer returns an error for invalid access tokens when using listFiles for a gated model

### ISSUE THREAD

**Issue by phudtran** [huggingface/huggingface.js#945]
## Issue

Hi, not sure if this is the correct place to post this issue since it seems to be an internal api change.

Previously, listing files in a gated repo using listFiles with an invalid or undefined access token would throw an authentication error. Now the authentication error is not thrown until an attempt is made to download one of the gated files with said invalid token.

Would it be possible/within scope to add some kind of hint to specify whether or not the user provided access token used in listModels/listFiles is able to access the gated model.

## Example

<img width="720" alt="image" src="https://github.com/user-attachments/assets/a90a0828-7f1a-4ac2-ac5c-e1662eb12ec2">

For example, when going to https://huggingface.co/google/gemma-2b/tree/main, I am immediately informed whether or not I have access to the gated model.

## Use case

Jellybox allows users to input their access token to download gated models after accepting the model conditions.

<img width="720" alt="image" src="https://github.com/user-attachments/assets/72190be0-2e0f-4e75-82f2-7a6389a79cd3">

**Comment by coyotte508** [comment]
Hi @phudtran 

The thing is that it's now possible to list gated files with the API.

If you want to check if a model is gated, I just created a PR adding `modelInfo`: https://github.com/huggingface/huggingface.js/pull/946

You can then do:

```ts
const info = await modelInfo({name: "some-model"});
console.log(info.gated);
```

**Comment by coyotte508** [comment]
Available in `@huggingface/hub@0.18.0` :)

**Comment by coyotte508** [comment]
And `await checkRepoAccess({repo: <model>, accessToken: "..."})` is available in `0.18.1`

**Comment by phudtran** [comment]
> And `await checkRepoAccess({repo: <model name>, accessToken: "..."})` is available in `0.18.1`: https://github.com/huggingface/huggingface.js/pull/947

Got it! Thank you!
